packages/core-transaction-pool/src/memory.ts

```diff
@@ -48,21 +56,21 @@ export class Memory {
         return this.all;
     }
 
-    public getExpired(maxTransactionAge: number): Interfaces.ITransaction[] {
+    public getExpired(): Interfaces.ITransaction[] {
         if (!this.byExpirationIsSorted) {
-            this.byExpiration.sort((a, b) => a.expiresAt(maxTransactionAge) - b.expiresAt(maxTransactionAge));
+            this.byExpiration.sort((a, b) => a.transaction.data.expiration - b.transaction.data.expiration);
             this.byExpirationIsSorted = true;
         }
 
-        const now: number = Crypto.slots.getTime();
+        const currentHeight: number = app.resolve("blockchain").getLastHeight();
```

Comment:
```suggestion
        let currentHeight: number;
        if (app.has("blockchain")) {
            currentHeight = app.resolvePlugin<Blockchain.IBlockchain>("blockchain").getLastHeight();
        } else {
            const { data }: Interfaces.IBlock = await app
                .resolvePlugin<Database.IDatabaseService>("database")
                .getLastBlock();

            currentHeight = data.height;
        }
```

Keep in mind that you will need to import the interfaces via `import { Blockchain, Database } from "@arkecosystem/core-interfaces";` and this change also requires the method to become `async` as the database call requires `await`.

Reply:
This causes an avalanche effect of turning all methods to `async`. That is not acceptable.

Instead I changed the code to try to query the current height from the state or blockchain in (an synchronous way), if they are initialized and if not - then handle "unknown current height".